packages/examples/cvat/exchange-oracle/src/handlers/job_creation.py

```diff
@@ -37,6 +35,12 @@
 from src.utils.assignments import parse_manifest
 from src.utils.logging import NullLogger, get_function_logger
 
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from logging import Logger
+
+    from src.core.manifest import TaskManifest
```

Comment:
What is the point of using type checking here? I don't think these imports have some special requirements.

Reply:
It's considered a good practice to keep imports used only in annotations in `if TYPE_CHECKING:` block, to clearly signify what code is actually used in runtime, not to mention reducing startup overhead and avoiding import cycles.

I wouldn't want to track these imports myself, but fortunately, `ruff` can take care of this automatically: by either moving imports unused in runtime to `if TYPE_CHECKING` block, or by moving import from this block, if import becomes used in runtime.

Reply:
From my knowledge, `TYPE_CHECKING` was added to support forward references and extra code for annotations specifically, then there such support was added in `from future import __annotations__`, now it makes sense to use it only for cyclic references between different files and heavy components (e.g. libraries with some initialization on import - hi, `tensorflow`), that are only used for annotations. It's hard to believe it's a considered a good practice to check all the code and search for imports that are only used for annotations.